tests/test_web/test_messaging.py

```diff
@@ -145,25 +145,25 @@ def test_get_ssl_domain_files_missing(mock_exists, cert_exists, key_exists, ca_e
     'missing_key', ('IIB_MESSAGING_CERT', 'IIB_MESSAGING_KEY', 'IIB_MESSAGING_CA')
 )
 @mock.patch('iib.web.messaging.os.path.exists')
-@mock.patch('iib.web.messaging.current_app')
-def test_get_ssl_domain_cert_config_not_set(mock_current_app, mock_exists, missing_key):
+def test_get_ssl_domain_cert_config_not_set(mock_exists, missing_key, app):
     mock_exists.return_value = True
-    mock_current_app.config = {
-        'IIB_MESSAGING_CERT': '/etc/iib/messaging.crt',
-        'IIB_MESSAGING_KEY': '/etc/iib/messaging.key',
-        'IIB_MESSAGING_CA': '/etc/iib/messaging.ca',
-    }
-    mock_current_app.config.pop(missing_key)
+    app.config.update(
+        {
+            'IIB_MESSAGING_CERT': '/etc/iib/messaging.crt',
+            'IIB_MESSAGING_KEY': '/etc/iib/messaging.key',
+            'IIB_MESSAGING_CA': '/etc/iib/messaging.ca',
+        }
+    )
+    app.config.pop(missing_key)
 
     ssl_domain = messaging._get_ssl_domain()
 
     assert ssl_domain is None
 
 
 @pytest.mark.parametrize('durable', (True, False))
-@mock.patch('iib.web.messaging.current_app')
-def test_json_to_envelope(mock_current_app, durable):
-    mock_current_app.config = {'IIB_MESSAGING_DURABLE': durable}
+def test_json_to_envelope(durable, app):
```

Comment:
I am missing the `app` variable. Do you mock it somehwere?

Reply:
`app` is a fixture defined on `tests/conftest.py` using the configuration from `iib.web.config.TestingConfig` which has the following config:

```
    DEBUG = True
    IIB_WORKER_USERNAMES = ['worker@DOMAIN.LOCAL']
    SQLALCHEMY_DATABASE_URI = f'sqlite:///{TEST_DB_FILE}'
    LOGIN_DISABLED = False
``` 

As stated in Observation 1 it's no longer possible to mock [Flask.current_app](https://stackoverflow.com/questions/57947935/mocking-current-app-in-flask-unit-pytest) so instead of doing it I'm using the fixture already made and including the configuration override which was being used in the original mock.

In this particular test case it requires the `app.configuration` to have the following key:

```
    {'IIB_MESSAGING_DURABLE': durable}
```
And `durable` is a boolean being parametrized as both `True` and `False`.

Since `IIB_MESSAGING_DURABLE` was not set in fixture's config it's safe to use and even if it was set the parametrize changes is from `True` to `False`, so no problem here.

Reply:
Just to clarify, `app` here is the fixture in place of the old argument `mock_current_app` which was defined like this:

```
@mock.patch('iib.web.messaging.current_app')
```

This is no longer possible, it throws the following error which happens on `mock.patch` decorator (I even tried to remove everything from test and let `assert True == True` and got the same result):

```
________________________________________________________________________________________________________ test_json_to_envelope[True] _________________________________________________________________________________________________________

args = (), keywargs = {'durable': True}

    @wraps(func)
    def patched(*args, **keywargs):
>       with self.decoration_helper(patched,
                                    args,
                                    keywargs) as (newargs, newkeywargs):

/usr/lib64/python3.8/unittest/mock.py:1322: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib64/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
/usr/lib64/python3.8/unittest/mock.py:1304: in decoration_helper
    arg = exit_stack.enter_context(patching)
/usr/lib64/python3.8/contextlib.py:425: in enter_context
    result = _cm_type.__enter__(cm)
/usr/lib64/python3.8/unittest/mock.py:1416: in __enter__
    if spec is None and _is_async_obj(original):
/usr/lib64/python3.8/unittest/mock.py:51: in _is_async_obj
    if hasattr(obj, '__func__'):
.tox/py38/lib/python3.8/site-packages/werkzeug/local.py:432: in __get__
    obj = instance._get_current_object()
.tox/py38/lib/python3.8/site-packages/werkzeug/local.py:554: in _get_current_object
    return self.__local()  # type: ignore
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def _find_app():
        top = _app_ctx_stack.top
        if top is None:
>           raise RuntimeError(_app_ctx_err_msg)
E           RuntimeError: Working outside of application context.
E           
E           This typically means that you attempted to use functionality that needed
E           to interface with the current application object in some way. To solve
E           this, set up an application context with app.app_context().  See the
E           documentation for more information.

.tox/py38/lib/python3.8/site-packages/flask/globals.py:47: RuntimeError

```

Reply:
This approach was already used in `test_send_messages`.
Question for discussion: Do we want to have these comments everywhere where `app` is used? 
Would not be sufficient just to mention it in commit message but without the comments? 
I can see that `app` is used almost everywhere (`test_messaging.py`, `test_api_v1.py` ... ) without this comment.

Reply:
I will leave it up to you. I haven't been a part of the developer team since the IIB started so I couldn't comment on this. I would like to approach this as an improvement of code documentation for the newcommers